### PR TITLE
test: add direct unit tests for MerkleTree class

### PR DESCRIPTION
**Pull request by 0xPlayerOne**
## Summary

Adds 12 real behavioral tests for the `MerkleTree` class (`src/scripts/merkle-distributor/helpers/merkle-tree.ts`), directly exercising its internal logic — construction, sorting, deduplication, proof generation and verification, and edge cases.

## Changes

- **`bun-tests/merkle-tree.test.ts`** — 12 `bun:test` cases covering:
  - Single-element tree (root = element)
  - Two-element tree (proofs generated)
  - Combined hash function
  - Sorting and deduplication
  - Empty tree error
  - Non-existent element error
  - Multi-element proof structure + manual verification
  - Determinism
  - Input-order independence (sorting normalizes)
  - Odd-count handling (last element pairs with self)
  - Undefined pair handling
  - Proof hex output format

## Risk Assessment

**LOW.** Pure TS test file, no contract changes. All 21 bun:test tests pass (9 existing + 12 new). Build OK (178 Solidity files). Type-check OK. Lint OK.

## Test Results

```
21 pass
0 fail
58 expect() calls
Ran 21 tests across 3 files. [42.00ms]
```

### CODE REVIEW

**Review by 0xPlayerOne — COMMENTED** [review]
Reviewed by Hermes Agent. No blocking issues found. Tests pass, CI is green, and lint/format/type-check are clean. See inline note on the  type edge case.

**Comment by 0xPlayerOne** [comment]
## Code Review Summary

**Verdict: Comment** (self-approval blocked by GitHub)

### ✅ Looks Good
- 12 new `bun:test` cases for `MerkleTree` all pass (21/21 total)
- Pure TS test addition, no contract changes
- `format:check`, `lint:ts`, `type-check`, and CI checks green
- Tests cover single-element, two-element, sorting/dedup, empty tree error, non-existent element error, multi-element proofs, determinism, input-order independence, odd-count handling, `combinedHash` undefined-input behavior, and proof hex formatting

### 💡 Suggestion (existing production code, not introduced by this PR)
- Inline comment left on `bun-tests/merkle-tree.test.ts` regarding `@ts-expect-error` around `combinedHash(a, undefined)` — the production implementation (`src/scripts/merkle-distributor/helpers/merkle-tree.ts`) accepts undefined buffers at runtime but is typed strictly as `Buffer`. This is a latent type-safety gap, but the test itself does a good job documenting the current runtime behavior.

No blocking issues. Ready to merge once an independent reviewer approves.

---
*Reviewed by Hermes Agent*

**Comment by 0xPlayerOne** [comment]
## Code Review Summary

**Verdict: Comment** (self-approval blocked by GitHub)

### Looks Good
- 12 new bun:test cases for MerkleTree all pass (21/21 total)
- Pure TS test addition, no contract changes
- format:check, lint:ts, type-check, and CI checks green
- Tests cover single-element, two-element, sorting/dedup, empty tree error, non-existent element error, multi-element proofs, determinism, input-order independence, odd-count handling, combinedHash undefined-input behavior, and proof hex formatting

### Suggestion (existing production code, not introduced by this PR)
- Inline comment left on bun-tests/merkle-tree.test.ts regarding type suppression around combinedHash(a, undefined). The production implementation accepts undefined buffers at runtime but is typed strictly as Buffer. This is a latent type-safety gap; test documents current runtime behavior well.

No blocking issues. Ready to merge once an independent reviewer approves.

---
*Reviewed by Hermes Agent*

File: bun-tests/merkle-tree.test.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    const rootBuf = tree.getRoot();
+    for (const el of elements) {
+      const proofBufs = tree.getProof(el);
+      expect(verifyProof(el, proofBufs, rootBuf)).toBe(true);
+    }
+  });
+
+  it('produces deterministic roots for the same input', () => {
+    const elements = [Buffer.from('alpha'), Buffer.from('beta'), Buffer.from('gamma')];
+    const tree1 = new MerkleTree(elements);
```
Comment (0xPlayerOne) [inline bun-tests/merkle-tree.test.ts:90]: Observation: This test documents runtime behavior where combinedHash(first, second) accepts undefined buffers and returns the non-null one. The production method is typed as combinedHash(first: Buffer, second: Buffer), so this currently requires @ts-expect-error to bypass the type checker. Not a blocker for this PR — the test is valuable — but consider widening the type signature in src/scripts/merkle-distributor/helpers/merkle-tree.ts to Buffer | undefined if this behavior is intentional.
